// marmiton-api/dist/lib/marmiton-api.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RECIPE_TYPE = exports.RECIPE_DIFFICULTY = exports.RECIPE_PRICE = exports.MarmitonQueryBuilder = exports.searchRecipes = exports.MarmitonError = void 0;
const marmiton_query_builder_1 = require("./components/marmiton-query-builder");
Object.defineProperty(exports, "MarmitonQueryBuilder", { enumerable: true, get: function () { return marmiton_query_builder_1.MarmitonQueryBuilder; } });
const recipe_enums_1 = require("./components/recipe-enums");
Object.defineProperty(exports, "RECIPE_DIFFICULTY", { enumerable: true, get: function () { return recipe_enums_1.RECIPE_DIFFICULTY; } });
Object.defineProperty(exports, "RECIPE_PRICE", { enumerable: true, get: function () { return recipe_enums_1.RECIPE_PRICE; } });
Object.defineProperty(exports, "RECIPE_TYPE", { enumerable: true, get: function () { return recipe_enums_1.RECIPE_TYPE; } });
const recipes_parser_1 = require("./components/recipes-parser");
class MarmitonError extends Error {
}
exports.MarmitonError = MarmitonError;
const BASE_URL = 'https://www.marmiton.org';
const ENDPOINTS = {
    query: () => `${BASE_URL}/recettes/recherche.aspx`,
};
// Number of recipes per page (this isn't constant)
const RECIPES_PER_PAGE = 13;
const DEFAULT_OPTIONS = {
    limit: RECIPES_PER_PAGE,
};
/**
 * Search for recipes within marmiton.com
 * @param qs querystring to use. This can be generated with {@link MarmitonQueryBuilder}
 * @param opt
 */
async function searchRecipes(qs, ingredients, // Ajoutez ce paramètre
opt) {
    const options = Object.assign(DEFAULT_OPTIONS, opt);
    const recipes = [];
    let url = `${ENDPOINTS.query()}?${qs}`;
    // Si des ingrédients sont fournis, modifiez l'URL pour inclure ces ingrédients
    if (ingredients && ingredients.length) {
        const ingredientsQueryString = ingredients.map(ingredient => encodeURIComponent(ingredient)).join('%2C-');
        url += `&aqt=${ingredientsQueryString}`;
    }
    for (let i = 1; recipes.length < options.limit; i++) {
        url += `&page=${i + 1}`;
        const request = await fetch(url);
        if (request.status !== 200)
            break;
        const htmlBody = await request.text();
        recipes.push(...(await recipes_parser_1.RecipesParser.parseSearchResults(htmlBody, BASE_URL)));
    }
    return recipes.slice(0, options.limit);
}
exports.searchRecipes = searchRecipes;
